### ISSUE TITLE
How to make non-ready pods discoverable by the plugin?

### ISSUE THREAD

**Issue by akvone** [hazelcast/hazelcast-kubernetes#56]
If two pods are initialized in parallel they don't see each other (because the Readiness Probe returns false) and form two separate clusters. Is there any way to prevent this behavior?

**Comment by googlielmo** [comment]
Hello @akvone thanks for reporting this issue, could you please indicate the version of the plugin that you're using? Is it the current master or 1.0.0?

This issue looks very similar to #33.

There's a discussion you might want to check on #16 and #33 also touching the issue of readiness checks. The first one was fixed by #30.
We believe that #30 also fixes #33, as we could not reproduce it; but we would be happy to be informed if that's not the case.

Cheers.

**Comment by googlielmo** [comment]
> Is there any way to prevent this behavior?

A simple workaround is to disable the readiness check in your pod definitions. The default one will suffice.

See for example https://github.com/hazelcast/hazelcast-docker/blob/master/hazelcast-kubernetes/hazelcast-service.yaml

**Comment by akvone** [comment]
We use 0.9.3 right now, but the same issue exists in 1.0.0 as well.
Unfortunately, we can't disable the readiness check as in that case load balancer will send requests on non-initialized pods.

As I got it, 1.0.1-SNAPSHOT contains the fixing commit. Could you please give an approximate date of 1.0.1 release?

**Comment by googlielmo** [comment]
We don't have an exact release date but it should be pretty soon.

That said, I would advise testing with the current snapshot so that if it does not work we could go through it and fix it before the release.
If it works, you could even continue using the snapshot until the release, as it is pretty stable.

I have some questions on the issue you are experiencing.
When you say that the pods "form two separate clusters", you have only two pods right? And do they start up initially as two clusters but then they  _eventually_ discover one another and form a cluster? What happens, say, after 10 minutes?

**Comment by akvone** [comment]
I've made some tests on two versions (0.9.3, 1.0.0) and got identical results. Let me show you:

There are logs I see on each pod: 

|   Time   |               Pod 1               |               Pod 2               |
|:--------:|:---------------------------------:|:---------------------------------:|
| 13:16:32 |  Spring Boot start initialization |                                   |
| 13:16:52 |                                   |  Spring Boot start initialization |
| 13:17:50 |            1 in cluster           |                                   |
| 13:17:51 | Spring Boot finish initialization |                                   |
| 13:17:58 |                                   |            1 in cluster           |
| 13:17:59 |                                   | Spring Boot finish initialization |
| 13:22:57 |            2 in cluster           |                                   |
| 13:22:57 |                                   |              2 cluser             |

So there is 5 min delay (13:17:59-13:22:57) during which both pods get the same requests. This breaks our business logic and we would like to fix it.

I'm checking 1.0.1-SNAPSHOT right now.

**Comment by googlielmo** [comment]
Okay, so the nodes eventually discover each other but that is not what you want.
Would #37 fix this for you, maybe? See the discussion there, I think it's a very similar use case.

Let me know how it went with 1.0.1-SNAPSHOT!

**Comment by googlielmo** [comment]
Hi @akvone, there is a new PR (#57) that I think may solve this issue, would you please have a look?
If it does, we could close this issue when we merge that. Thank you!

**Comment by akvone** [comment]
Official 1.0.1-SNAPSHOT doesn't work. 
But version with commit from PR (#57) works well. That's that we want.

Now the question about 1.0.1 release is really topical. I'll be glad to get any news about this.

Thank you!

**Comment by googlielmo** [comment]
That's great news @akvone, thank you for confirming this!
Keep an eye on this repo for the upcoming release.

**Comment by akvone** [comment]
I'm sorry to bother you again. But I'd like to ask you for an advice. 

We are currently at the crossroad between waiting for the official 1.0.1 release and working around the issue by ourselves.
Is there any chance for 1.0.1 to be released in a month or less? Or is it clear already that the release date is much later in the roadmap?

Many thanks!

**Comment by googlielmo** [comment]
@akvone We should have a new release in the next few days, not months!
Cheers,
Guglielmo

**Comment by burakcelebi** [comment]
Fixed by #57

**Comment by akvone** [comment]
Well, we have successfully included 1.1.0 in our project. Thanks a lot!

But I've just found minor discrepancy between property definition (resolve-not-ready-addresses) javadoc and code. There it is:
https://github.com/hazelcast/hazelcast-kubernetes/blob/833f232684d87fc1b44c0e6968d9abae4e21aaf5/src/main/java/com/hazelcast/kubernetes/KubernetesProperties.java#L89

Should I create an issue?

**Comment by hasancelik** [comment]
@akvone new seperate issue would be great 👍
